am/config.py

```diff
@@ -1,24 +1,63 @@
 __all__ = ["acoustic_model_config"]
 
+from dataclasses import dataclass
+from typing import List, Literal, Tuple, Union
 import i6_core.rasr as rasr
 
+from sisyphus import tk
+
+TdpType = Union[float, Literal["infinity"]]
+
+
+@dataclass
+class TdpValues:
+    loop: TdpType
+    forward: TdpType
+    skip: TdpType
+    exit: TdpType
+
 
 def acoustic_model_config(
-    state_tying="monophone",
-    states_per_phone=3,
-    state_repetitions=1,
-    across_word_model=True,
-    early_recombination=False,
-    tdp_scale=1.0,
-    tdp_transition=(3.0, 0.0, 3.0, 2.0),
-    tdp_silence=(0.0, 3.0, "infinity", 6.0),
-    tying_type="global",
-    nonword_phones="",
-    tdp_nonword=(0.0, 3.0, "infinity", 6.0),
-):
+    state_tying: str = "monophone",
+    states_per_phone: int = 3,
+    state_repetitions: int = 1,
+    across_word_model: bool = True,
+    early_recombination: bool = False,
+    tdp_scale: float = 1.0,
+    tdp_transition: Union[TdpValues, Tuple[TdpType, ...]] = (3.0, 0.0, 3.0, 2.0),
+    tdp_silence: Union[TdpValues, Tuple[TdpType, ...]] = (0.0, 3.0, "infinity", 6.0),
+    tying_type: Literal["global", "global-and-nonword"] = "global",
+    nonword_phones: Union[str, List[str]] = "",
+    tdp_nonword: Union[TdpValues, Tuple[TdpType, ...]] = (0.0, 3.0, "infinity", 6.0),
+    state_tying_file: Union[str, tk.Path] = "",
```

Comment:
can we make this
```suggestion
    state_tying_file: Optional[tk.Path] = None,
```
disallowing `str`. 

And below check for `state_tying_file is not None`